pyteal/ast/subroutine.py

```diff
@@ -76,37 +78,176 @@ def __init__(
         self.declaration: Optional["SubroutineDeclaration"] = None
         self.__name = self.implementation.__name__ if nameStr is None else nameStr
 
+    @staticmethod
+    def check_if_ABI_class(arg_type: Type) -> bool:
+        return (
+            isclass(arg_type)
+            and arg_type is not abi.BaseType
+            and issubclass(arg_type, abi.BaseType)
+        )
+
+    @staticmethod
+    def check_if_ABI_annotation(abi_anno: Any) -> bool:
+        try:
+            abi.type_spec_from_annotation(abi_anno)
+            return True
+        except TypeError:
+            return False
+
+    @staticmethod
+    def _check_ABI_output(annotations: Dict[str, type], sig: Signature) -> None:
```

Comment:
Why can't we use the same approach as before to look at the return type and see if it has an annotation present? If present, then use that rather than ask the user to use `output`  as a keyword. Seems like that is a more consistent approach.

For a similar reason, the approach taken in my PR #249 (using `input_types`) is less preferable than the approach taken here.